src/main.ts

```diff
@@ -19,12 +19,12 @@ const main = async () => {
 
   await pulumiCli.downloadCli(config.options.pulumiVersion);
 
-  if (environmentVariables.PULUMI_ACCESS_TOKEN !== '') {
-    core.debug(`Logging into Pulumi`);
-    await pulumiCli.run('login');
-  } else if (config.cloudUrl) {
+  if (config.cloudUrl) {
     core.debug(`Logging into ${config.cloudUrl}`);
     await pulumiCli.run('login', config.cloudUrl);
+  } else {
+    core.debug(`Logging into Pulumi`);
+    await pulumiCli.run('login');
```

Comment:
I think we still need the first conditional guard as an `else if` branch to preserve the original behavior, no?

Reply:
`else if` to check whether `PULUMI_ACCESS_TOKEN` is set? If so... I think you're right, because the `pulumi login` command would block indefinitely on the prompt for a token if you were trying to login to a Pulumi backend (either cloud- or self-hosted). I can work on a fix in the morning.

Reply:
On that note, are there any scenarios under which the remainder of the script can do any useful work if `pulumi login` hasn't been called?  If not, I'd think we'd want to throw an exception in an `else` block rather than wait for later commands to fail.

Reply:
Although thinking on this a little more, the "login isn't necessary" case probably encompasses all the non-Pulumi-enterprise backends, in which case we definitely want to preserve the existing behavior of just continuing without logging in if neither condition matches.